File: android-vinetrack/app/src/main/java/com/rork/vinetrack/data/auth/BiometricEnrollmentEligibility.kt
package com.rork.vinetrack.data.auth

/** Shared decision for a fresh login or a restored session after the main shell is ready. */
object BiometricEnrollmentEligibility {
    fun shouldOffer(isInMainShell: Boolean, supportsDeviceAuth: Boolean, isEnabled: Boolean, hasPrompted: Boolean): Boolean =
        isInMainShell && supportsDeviceAuth && !isEnabled && !hasPrompted

    fun shouldLock(hasRestoredSession: Boolean, isEnabled: Boolean): Boolean =
        hasRestoredSession && isEnabled
}

File: ios/VineTrack/App/BiometricEnrollmentEligibility.swift
import Foundation

/// Decides whether a restored or freshly signed-in session may show the one-time offer.
nonisolated enum BiometricEnrollmentEligibility {
    static func shouldOffer(isInMainShell: Bool, supportsDeviceAuth: Bool, isEnabled: Bool, hasPrompted: Bool) -> Bool {
        isInMainShell && supportsDeviceAuth && !isEnabled && !hasPrompted
    }

    static func shouldLock(hasRestoredSession: Bool, isEnabled: Bool, supportsDeviceAuth: Bool) -> Bool {
        hasRestoredSession && isEnabled && supportsDeviceAuth
    }
}
